include/ScratchViews.h

```diff
@@ -223,32 +248,31 @@ void fill_pre_req_data(
   ScratchViews& prereqData)
 {
   int nodesPerElem = topo.num_nodes();
-  const stk::mesh::Entity* nodes = nullptr;
 
   // extract master elements
   MasterElement *meSCS = dataNeeded.get_cvfem_surface_me();
   MasterElement *meSCV = dataNeeded.get_cvfem_volume_me();
 
   const FieldSet& neededFields = dataNeeded.get_fields();
-  for(const stk::mesh::FieldBase* fieldPtr : neededFields) {
-    stk::mesh::EntityRank fieldEntityRank = fieldPtr->entity_rank();
+  for(const FieldInfo& fieldInfo : neededFields) {
+    stk::mesh::EntityRank fieldEntityRank = fieldInfo.field->entity_rank();
+    unsigned scalarsPerEntity = fieldInfo.scalarsPerEntity;
     if (fieldEntityRank==stk::topology::ELEM_RANK) {
-      SharedMemView<double*>& shmemView = prereqData.get_scratch_view_1D(*fieldPtr);
+      SharedMemView<double*>& shmemView = prereqData.get_scratch_view_1D(*fieldInfo.field);
       unsigned len = shmemView.dimension(0);
-      double* fieldDataPtr = static_cast<double*>(stk::mesh::field_data(*fieldPtr, elem));
+      double* fieldDataPtr = static_cast<double*>(stk::mesh::field_data(*fieldInfo.field, elem));
       for(unsigned i=0; i<len; ++i) {
         shmemView(i) = fieldDataPtr[i];
       }
     }
     else if (fieldEntityRank == stk::topology::NODE_RANK) {
-      SharedMemView<double*>& shmemView1D = prereqData.get_scratch_view_1D(*fieldPtr);
-      if (shmemView1D.dimension(0) > 0) {
-        gather_elem_node_field(*fieldPtr, elem, nodesPerElem, bulkData.begin_nodes(elem), shmemView1D);
+      if (scalarsPerEntity == 1) {
```

Comment:
This works now for scalar and vectors (sized by nDim). I think that in some rare cases, we have tensor quantities.